core/sail/memory/src/main/java/org/eclipse/rdf4j/sail/memory/MemoryStore.java

```diff
@@ -201,6 +213,70 @@ public long getSyncDelay() {
 	public synchronized EvaluationStrategyFactory getEvaluationStrategyFactory() {
 		if (evalStratFactory == null) {
 			evalStratFactory = new StrictEvaluationStrategyFactory(getFederatedServiceResolver());
+			evalStratFactory.addOptimizer((strategy, tripleSource, evaluationStatistics) -> {
+				return new QueryOptimizer() {
```

Comment:
If `addOptimizer` accepts a `QueryOptimizer` (instead of a `QueryOptimizerFunctionalInterace`) this lambda becomes unnecessary.

Reply:
Won't get access to any of the variables then. Eg. strategy, tripleSource and evaluationStatistics